### DISCUSSION TITLE
Does red support multi-language cogs?

### DISCUSSION THREAD

**Discussion by zephiowl** [Cog-Creators/Red-DiscordBot#5706] [Q&A] [answered]
Hey so I see that red is written with python; but I was wondering (I'm relatively new to programming) If it was at all possible to write a red cog in another language outside of python; say java; js; rust; etc. If so how would i go about doing that? as its something I have interest in; if not I'll use python its just something that seems like a good idea if possible.

**Comment by Flame442** [comment, marked as answer]
Red is a python bot, and cogs interface with the discord.py cog system to extend the bot's functionality. While it is *technically* possible to create a cog which allows running code written in another language (and by proxy, code additional functionality in another language), some python code would be required to build that interface. A few such cogs exist, I know Neuro [has one for a few different languages](https://github.com/NeuroAssassin/Toxic-Cogs/tree/master/esolang). It is far from the easiest way to write a cog, however, so I would still advise writing most cogs in python.

**Reply by zephiowl** [reply]
Thanks that's exactly what I was after! I'm just a little concerned about the future of red due to the lack of development with discord.py I read through and saw that some other platforms like https://github.com/serenity-rs/serenity is supposed to have support for discords new requirements. I run a discord bot built with red here and some internal cogs. Which is a reason I was asking if it was possible for multi-lang support. But thank you I'll be sure to come back if i have any more questions!

**Reply by Flame442** [reply]
> I'm just a little concerned about the future of red due to the lack of development with discord.py

Discord.py's creator (Danny) has come back to developing, so there is no worry about the future of Red (you can check the main branch of [that repo](https://github.com/Rapptz/discord.py) to see for yourself). Even if that was not the case, Red would have been fine. We're taking our time (the time discord has graciously given us) to make sure we are creating the best experience for our users, not trying to rush into it.

**Reply by zephiowl** [reply]
> Discord.py's creator (Danny) has come back to developing. 

Ahh I wasn't aware of that; last I knew he stopped working on it due to a lack of motivation! Ty for letting me know. 

> Even if that was not the case, Red would have been fine. We're taking our time (the time discord has graciously given us) to make sure we are creating the best experience for our users,

Well that's really awesome to see; I was kind of afraid that once discord.py got discontinued this project would follow suit; requiring all of us to essentially start from scratch again. As I've heard horror story's from other devs of very similar events happening to other systems.

I appreciate all of the hard work that's been put into red and I really hope it only continues to grow and prosper.
